docs/reference/search/search-your-data/paginate-search-results.asciidoc

```diff
@@ -28,29 +20,178 @@ GET /_search
 }
 ----
 
-By default, you cannot page through more than 10,000 documents using the `from`
-and `size` parameters. This limit is set using the
-<<index-max-result-window,`index.max_result_window`>> index setting.
+By default, you cannot use `from` and `size` to page through more than 10,000
+hits. Using `from` and `size` requires each searched shard to load hits for
+previous pages into memory. This can significantly increase memory and CPU
+usage, resulting in degraded performance or node failures.
+
+If you need to page through more than 10,000 hits, use the
+<<search-after,`search_after`>> parameter. If needed, you can use the
+<<index-max-result-window,`index.max_result_window`>> index setting to change
+the 10,000 hit safeguard.
+
+WARNING: {es} uses Lucene's internal doc IDs as tie-breakers. These internal doc
+IDs can be completely different across replicas of the same data. When paging
+search hits, you might occasionally see that documents with the same sort values
+are not ordered consistently.
+
+[discrete]
+[[search-after]]
+=== Search after
+
+You can use the `search_after` parameter to retrieve the next page of hits
+using a set of <<sort-search-results,sort values>> from the previous page.
+
+Using `search_after` requires multiple search requests with the same `query` and
+`sort` values. If a <<near-real-time,refresh>> occurs between these requests,
+the order of your results may change, resulting in inconsistent pages. To
+prevent this, you can create a <<point-in-time-api,point in time (PIT)>> to
+preserve the current index state for a specified period.
+
+IMPORTANT: We don't recommend creating PITs for each user request. Each PIT
+requires additional disk space and file handles. If you want to paginate search
+results for user requests, skip this step and use `search_after` without a PIT.
+
+[source,console]
+----
+POST /my-index-000001/_pit?keep_alive=1m
+----
+// TEST[setup:my_index]
+
+The API returns a PIT ID.
+
+[source,console-result]
+----
+{
+  "id": "46ToAwMDaWR4BXV1aWQxAgZub2RlXzEAAAAAAAAAAAEBYQNpZHkFdXVpZDIrBm5vZGVfMwAAAAAAAAAAKgFjA2lkeQV1dWlkMioGbm9kZV8yAAAAAAAAAAAMAWICBXV1aWQyAAAFdXVpZDEAAQltYXRjaF9hbGw_gAAAAA=="
+}
+----
+// TESTRESPONSE[s/"id": "46ToAwMDaWR4BXV1aWQxAgZub2RlXzEAAAAAAAAAAAEBYQNpZHkFdXVpZDIrBm5vZGVfMwAAAAAAAAAAKgFjA2lkeQV1dWlkMioGbm9kZV8yAAAAAAAAAAAMAWICBXV1aWQyAAAFdXVpZDEAAQltYXRjaF9hbGw_gAAAAA=="/"id": $body.id/]
+
+To get the first page of results, submit a search request with a `sort`
+argument. If using a PIT, include the PIT ID as the `pit.id` argument.
+
+IMPORTANT: We recommend you include a tiebreaker field in your `sort`. This
+tiebreaker field should contain a unique value for each document. If you don't
+include a tiebreaker field, your paged results could miss or duplicate hits.
```

Comment:
For reviewers:

The original admon suggested using the `set` ingest processor to create a tiebreaker field containing the doc's `_id` value. However, you can't use ingest processors to access autogen docs IDs.

Is it worth explicitly mentioning enabling `indices.id_field_data.enabled` and sorting on `_id` here?

Reply:
No this should be avoided and in 8.0 this will probably be forbidden. We have a follow up for PIT to add a simple tiebreaker in https://github.com/elastic/elasticsearch/issues/56828. We want to have it soon so that users don't need to worry about creating a doc value field for this purpose. That said, I think that the documentation should be updated when we have the new feature but the new shape looks great. I think we should proceed with this update to the docs and add the virtual tiebreaker part when it's ready.

Reply:
Sounds good. Thanks for the context @jimczi.